webapp/src/app/admin/controls/control.service.ts

```diff
@@ -0,0 +1,167 @@
+import { Injectable } from '@angular/core';
+import { ACLApi, SDKModels, RoleApi } from '../../shared/sdk/services/custom';
+import { Subscription } from 'rxjs/Subscription';
+import { FormService } from '../../ui/form/ui-form.service';
+import { Observable } from 'rxjs/Observable';
+import { sortBy } from 'lodash';
+
+@Injectable()
+export class ControlService {
+
+  private subscriptions: Subscription[] = new Array<Subscription>();
+  models: any;
+  roles: any;
+
+  constructor(
+    private formService: FormService,
+    private aclApi: ACLApi,
+    private sdkModels: SDKModels,
+    private roleApi: RoleApi
+  ) {
+    this.getModels();
+    this.getRoles();
+  }
+
+  getCardButtons() {
+    return {
+      class: 'btn btn-secondary btn-block float-right',
+      icon: 'plus',
+      text: 'Create'
+    };
+  }
+
+  getTableHeaders() {
+    return [
+      'Model',
+      'Property',
+      'Access Type',
+      'Permission',
+      'Principal Type',
+      'Principal ID',
+      'Actions'
+    ];
+  }
+
+  getFormConfig(formType: string) {
+    return {
+      fields: this.getFormFields(formType),
+      showCancel: true,
+      action: formType === 'create' ? formType : 'update',
+    };
+  }
+
+  getModels() {
+    const models = this.sdkModels.getAll();
+    this.models = sortBy(Object.keys(models));
+  }
+
+  getRoles() {
+    this.subscriptions.push(this.roleApi.find().subscribe((roles: any) => (this.roles = roles)));
+  }
+
+  getFormFields(formType: string) {
+    let models = [];
+    let roles = [];
+    const defaultRoles = ['$authenticated', '$everyone', '$owner', '$unauthenticated'];
```

Comment:
should not this be defined as a class parameter? so it is declared just once... Here every time you call the getFormFields method you redeclare the const defaultRoles which is wrong, a const should not be re-declared